pmm-app/src/pmm-qan/panel/components/Filters/Filters.tools.ts

```diff
@@ -1,11 +1,14 @@
 import { getTemplateSrv } from '@grafana/runtime';
-import { FILTERS_GROUPS } from './Filters.constants';
 
-export const getSelectedCheckboxes = (filters) => FILTERS_GROUPS.map((group) => filters[group.dataKey])
-  .filter(Boolean)
-  .map((item) => item.name)
-  .reduce((acc, item) => acc.concat(item), [])
-  .some((item) => item.checked);
+export const getSelectedCheckboxes = (filters, filtersGroups) => {
+  const selectedFilters = filtersGroups.map((group) => filters[group.dataKey])
```

Comment:
what if `group` is null/undefined or there is no item in filters for `group.dataKey`

Reply:
This `filtersGroups` is based on the const FILTERS_GROUPS with the addition of groups from the comment of the selected query (if there is one), so it always has groups from the constant.